sympy/solvers/polysys.py

```diff
@@ -429,3 +439,428 @@ def solve_triangulated(polys, *gens, **args):
+    >>> a, b, c, x = symbols('a b c x')
+    >>> p1 = Poly((a - 1)*(x - 2), x, domain=ZZ[a,b,c])
+    >>> p2 = Poly((b - 3)*(x - 2), x, domain=ZZ[a,b,c])
+    >>> p3 = Poly(c, x, domain=ZZ[a,b,c])
+
+    The equation to be solved for x is ``x - 2 = 0`` provided either
+    of the two conditions on the parameters ``a`` and ``b`` is nonzero
+    and the constant parameter ``c`` should be zero.
+
+    >>> systems, constant_conds = factor_system_poly([p1, p2, p3])
+    >>> systems
+    [[(Poly(x - 2, x, domain='ZZ[a,b,c]'), (Poly(a - 1, x, domain='ZZ[a,b,c]'), Poly(b - 3, x, domain='ZZ[a,b,c]')))]]
+
+    >>> constant_conds
+    [Poly(c, x, domain='ZZ[a,b,c]')]
+
+    This is the core routine used by higher-level functions factor_system_cond,
+    factor_system, and factor_system_bool.It Returns empty systems list
+    ([]) when no solution exists and [[]] (list containing empty list) when any
+    value is a solution
+
+    See Also
+    ========
+
+    factor_system : Returns factors and solvability condition separately
+    factor_system_bool : Returns a Boolean combination representing the solution
+    factor_system_cond : Returns both factors and conditions
+    sympy.polys.polytools.factor : Factors a polynomial into irreducible factors
+                                   over the rational numbers
+
+    """
+    if not all(isinstance(poly, Poly) for poly in polys):
+        raise TypeError("polys should be a list of Poly instances")
+    if not polys:
+        return [[]], []
+    else:
+        base_domain = polys[0].domain
+        base_gens = polys[0].gens
+        if not all(poly.domain == base_domain and poly.gens == base_gens for poly in polys[1:]):
+            raise DomainError("All polynomials must have the same domain and generators")
+
+    constant_eqs = []
+    eqs_factors = []
+    conds_factor = defaultdict(list)
+
+    for poly in polys:
+        constant, factors_mult = poly.factor_list()
+
+        factors = [f for f, m in factors_mult]
+
+        if factors:
+            eqs_factors.append(factors)
+            if constant.is_zero is not False:
+                constp = constant.as_poly(base_gens, domain=base_domain)
+                for f in factors:
+                    if constp not in conds_factor[f]:
+                        conds_factor[f].append(constp)
+        elif constant.is_zero is True:
+            pass
+        elif constant.is_zero is False:
+            return ([], [])
+        else:
+            constp = constant.as_poly(base_gens, domain=base_domain)
+            constant_eqs.append(constp)
+
+    fac2conds = {f: tuple(conds) for f, conds in conds_factor.items()}
+
+    cnf = [[(f, fac2conds.get(f, ())) for f in eq] for eq in eqs_factors]
+
+    dnf = _cnf2dnf(cnf)
+
+    return dnf, constant_eqs
+
+
+def _has_common_variables(factor1, factor2):
+    """Helper function to check if two polynomial factors share any variables"""
+    return bool(set(factor1.gens).intersection(factor2.gens))
```

Comment:
The polynomials all have the same gens so I think that this is always going to be true. This should probably be checking the degrees of the polynomials.

Reply:
I agree, we do ensure that gens are same for them all. I should've noticed that earlier. 

Not sure how this should be structured as of now though.

Reply:
It should use `degree_list`:
```python
In [64]: p1 = Poly(x**2, [x, y])

In [65]: p2 = Poly(y, [x, y])

In [66]: p1
Out[66]: Poly(x**2, x, y, domain='ZZ')

In [67]: p2
Out[67]: Poly(y, x, y, domain='ZZ')

In [68]: p1.gens
Out[68]: (x, y)

In [69]: p2.gens
Out[69]: (x, y)

In [71]: p1.degree_list()
Out[71]: (2, 0)

In [72]: p2.degree_list()
Out[72]: (0, 1)
```
The polynomials have a symbol in common if:
```
In [74]: any(d1 and d2 for d1, d2 in zip(p1.degree_list(), p2.degree_list()))
Out[74]: False
```